File: gaia-sdk-javascript/src/graphql/request/type/CreatedSkillImpulse.ts

import {Skill} from "./Skill";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";

/**
 * Impulse which indicates the result of a create Skill impulse
 */
export class CreatedSkillImpulse extends Array<(_:VariableRegistry) => string> {

    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * the Skill instance
     */
    public data = (config: (_:Skill) => void) => this.push((registry) => {
        const entity = new Skill();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/CreatedSkillImpulse.py

from gaia_sdk.graphql.request.type.Skill import Skill

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreatedSkillImpulse(list):
    """
    Impulse which indicates the result of a create Skill impulse
    """

    def id(self):
        self.append(lambda x: "id")

    """
    the Skill instance
    """
    def data(self, config: Callable[['Skill'], None]):
        def callback(registry: VariableRegistry):
            entity = Skill()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
